doc/howto/write_an_alphabet/index.md

```diff
@@ -0,0 +1,149 @@
+`to_rank`, `assign_rank` and `alphabet_size` for our type, SeqAn automatically exposes them to the respective global
+functions `seqan3::to_rank`, `seqan3::assign_rank_to` and `seqan3::alphabet_size`.
+Note that `assign_rank` is only required for seqan3::WritableSemialphabet.
+\snippet dna2_semialphabet.cpp semialphabet
+
+As you can see from the `static_assert` our dna2 alphabet now models seqan3::Semialphabet and
+seqan3::WritableSemialphabet:
+\snippet dna2_semialphabet.cpp writable_semialphabet_concept
+
+With this we can automatically use some free functions on our dna2 alphabet. If you are interested in how this works,
+read our section on [Customisation points](\ref about_customisation). Here is an example:
+\snippet dna2_semialphabet.cpp free_functions
+
+## Alphabet
+
+Now that you have a feeling for concepts, have a look at seqan3::Alphabet and seqan3::WritableAlphabet and make
+your type also model these concepts. If you keep the suggested namings, SeqAn will automatically
+expose the members to the respective global functions that model the concept.
+
+\assignment{Excercise}
+1. Implement the member function `char to_char(void)` which returns the char 'S' or 'W' depending on the rank value.
+2. Implement the member function `dna2 & assign_char(char const)`.
+3. seqan3::WritableAlphabet additionally requires that the type has a member function `bool char_is_valid`
+   which can be used to determine whether a given character is an actual representation of the alphabet.
+   For the `dna2` alphabet 'S' and 'W' as well as 's' and 'w' should be valid, but nothing else.
+   <br>
+   Implement the static member function `bool char_is_valid(char const)`.
+\endassignment
+\solution
+\snippet dna2_alphabet.cpp writable_alphabet
+\endsolution
+
+At this point the seqan3::Alphabet concept should be modelled successfully and even seqan3::WritableAlphabet 
+is fine because we implemented `assign_char` and `char_is_valid`.
+\snippet dna2_alphabet.cpp writable_alphabet_concept
+
+## Shortcut: Alphabet base template
+
+In reality, you do not need to define all the functions you have learned in this exercise manually.
+Instead, you can inherit your type from seqan3::alphabet_base and just define the char-rank conversion
+in both directions. Read the documentation of seqan3::alphabet_base for details.
+\snippet dna2_inherit_base.cpp dna2
+
+# Further examples
+## Implementation as enum class
+
+This is an example of a minimal custom alphabet that provides implementations for all necessary customisation
+points. As an enum class it already models std::StrictTotallyOrdered.
+\snippet test/unit/alphabet/custom_alphabet_test.cpp my_alph
+
+## Implementation of a non-default-constructible class
+
+This is an example of a custom alphabet that is not default-constructible and that has a non-default overload for
+seqan3::char_is_valid_for.
+\snippet test/unit/alphabet/custom_alphabet2_test.cpp my_alph
+
```

Comment:
Describe the few things that are different and why!

Reply:
This is still open.

Reply:
oops... I had just applied it for the first snippet. 
I have provided some explanations above the last snippet now.